tremor-script/src/grammar.lalrpop

```diff
@@ -580,11 +582,20 @@ ModuleExprs: ExprsRaw<'input> = {
 
 ModuleExpr: ExprRaw<'input> = {
     Const => <>,
+    AggregateFnDecl => ExprRaw::AggregateFnDecl(<>),
     FnDecl => ExprRaw::FnDecl(<>),
     Intrinsic => ExprRaw::FnDecl(<>),
     Module => ExprRaw::Module(<>),
 }
 
+AggregateFnDecl: AggrFnDeclRaw<'input> = {
+    <doc:DocComment> <start:@L> "aggregate" <name:Ident> "of" <body:AggregateFnBody> "end" <end:@L> => AggrFnDeclRaw{start, end, name, body}
+}
+
+AggregateFnBody: AggrFnBodyRaw<'input> = {
+    <start:@L> "init" "(" ")" <init_body:Effectors> "aggregate" "(" <aggregate_args:FnArgs> ")" <aggregate_body:Effectors> "mergein" "(" <merge_args:FnArgs> ")" <merge_body:Effectors> "emit" "(" <emit_args:FnArgs> ")" <emit_body:Effectors> <end:@L> => AggrFnBodyRaw{start, end, init: InitDeclRaw(init_body), aggregate: AggregateDeclRaw(aggregate_args, aggregate_body), merge: MergeInDeclRaw(merge_args, merge_body), emit: EmitDeclRaw(emit_args, emit_body)}
```

Comment:
I think a newline would be nice somewhere in this (or being split out into sub rules)